src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,43 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOUR_COUNT_INDEX = 2;
+    private static final int INCOME_PER_HOUR_INDEX = 3;
+    private static final String LINE_SPLIT = " ";
+    private static final String DATE_SPLIT = "\\.";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Report for period ").append(dateFrom)
+                .append(" - ").append(dateTo);
+        LocalDate localDateFrom = getDateFromString(dateFrom);
+        LocalDate localDateTo = getDateFromString(dateTo);
+        for (int i = 0; i < names.length; i++) {
+            int salary = 0;
+            stringBuilder.append(System.lineSeparator()).append(names[i]).append(" - ");
```

Comment:
All magic numbers, strings, symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)